File: commons/javascript/utils_test.js
import get_minimize_data from './utils.mjs'

import data_1  from "../../test/resource/sample_sbom_1.json" assert {type: 'json'}
import data_2  from "../../test/resource/sample_sbom_2.json" assert {type: 'json'}

var data_3 = {
    "licenses": [
        {
            "license": {
                "name": "BSD-3-Clause"
            }
        }
    ]
}

var parsedJsonString_1 = get_minimize_data(data_1);
var parsedJsonString_2 = get_minimize_data(data_2);
var parsedJsonString_3 = get_minimize_data(data_3);

var requiredJsonString = '{"licenses":[{"license":{"name":"BSD-3-Clause"}}]}';

console.log(parsedJsonString_1 == requiredJsonString);
console.log(parsedJsonString_2 == requiredJsonString);
console.log(parsedJsonString_3 == requiredJsonString);

File: commons/python/utils_test.py
import json
from utils import *


# Minimize data test case

f = open('./../../test/resource/sample_sbom_1.json')
data_1 = json.load(f)
parsedJsonString_1 = get_minimize_data(data_1)
assert parsedJsonString_1 == '{"licenses":[{"license":{"name":"BSD-3-Clause"}}]}'

f = open('../../test/resource/sample_sbom_2.json')
data_2 = json.load(f)
parsedJsonString_2 = get_minimize_data(data_2)
assert parsedJsonString_2 == '{"licenses":[{"license":{"name":"BSD-3-Clause"}}]}'

data_3 = {'licenses': [{"license": {"name": "BSD-3-Clause"}}]}
parsedJsonString_3 = get_minimize_data(data_3)
assert parsedJsonString_3 == '{"licenses":[{"license":{"name":"BSD-3-Clause"}}]}'

